node/overseer/overseer-gen/proc-macro/src/impl_builder.rs

```diff
@@ -107,46 +61,379 @@ pub(crate) fn impl_builder(info: &OverseerInfo) -> proc_macro2::TokenStream {
 		})
 		.collect::<Vec<_>>();
 
-	let generics = quote! {
-		< S, #( #baggage_generic_ty, )* >
-	};
-	let where_clause = quote! {
+	// A helper structure to carry fields of a builder to produce setters
+	struct BuilderFieldHelper {
+		// Used to distinguish baggage and subsystems
+		is_subsystem: bool,
+		// Valid for baggage as all subsystems are generics
+		is_generic: bool,
+		// Field name
+		name: Ident,
+		// Field type (either generic or not)
+		field_ty: TokenStream,
+		// Type to use in generics (like __FieldName)
+		placeholder_type: TokenStream,
+		// Used to constrain setter generics
+		where_clause: Option<TokenStream>,
+		// Specific initialised type, like OverseerFieldInit<T>
+		inited_ty: TokenStream,
+		// Specific uninitialised type, like OverseerFieldUninit<T>
+		uninited_ty: TokenStream,
+		// Order of the field in a target structure
+		order: usize,
+	}
+
+	let mut field_helpers = info.subsystems()
+		.iter()
+		.filter(|ssf| !ssf.wip)
+		.enumerate()
+		.map(|(idx, ssf)| {
+			let prefixed_ident = format_ident!("__{}",
+				ssf.name.to_string().to_class_case());
+			let consumes = &ssf.consumes;
+			let generic_ty = &ssf.generic;
+			BuilderFieldHelper {
+				is_subsystem: true,
+				is_generic: true,
+				name: ssf.name.clone(),
+				field_ty: quote!(#generic_ty),
+				placeholder_type: quote! { #prefixed_ident },
+				where_clause: Some(quote! {
+					#generic_ty : #support_crate ::Subsystem<#subsystem_ctx_name< #consumes >, #error_ty>
+				}),
+				inited_ty: quote! {
+					OverseerFieldInit<SubsystemFieldInitMethod<#generic_ty>>
+				},
+				uninited_ty: quote! {
+					OverseerFieldUninit<SubsystemFieldInitMethod<#generic_ty>>
+				},
+				order: idx
+			}
+		})
+		.collect::<Vec<_>>();
+	let last_subsystem_idx = field_helpers.len();
+	field_helpers.extend(info.baggage()
+		.iter()
+		.enumerate()
+		.map(|(idx, bgf)| {
+			let prefixed_ident = format_ident!("__{}", bgf.field_name
+				.to_string().to_class_case());
+			let baggage_ty = &bgf.field_ty;
+
+			BuilderFieldHelper {
+				is_subsystem: false,
+				is_generic: bgf.generic,
+				name: bgf.field_name.clone(),
+				field_ty: quote!(#baggage_ty),
+				placeholder_type: quote! { #prefixed_ident },
+				where_clause: None,
+				inited_ty: quote! {
+					OverseerFieldInit<#baggage_ty>
+				},
+				uninited_ty: quote! {
+					OverseerFieldUninit<#baggage_ty>
+				},
+				order: idx + last_subsystem_idx
+			}
+		})
+	);
+
+	let struct_common_generics = field_helpers
+		.iter()
+		.filter(|fh| fh.is_generic)
+		.map(|fh| &fh.field_ty)
+		.collect::<Vec<_>>();
+	let builder_common_generics = field_helpers
+		.iter()
+		.map(|fh| &fh.placeholder_type)
+		.collect::<Vec<_>>();
+	let where_clause_elts = field_helpers
+		.iter()
+		.filter_map(|fh| fh.where_clause.as_ref())
+		.collect::<Vec<_>>();
+
+	let builder_where_clause = quote! {
 		where
-			S: #support_crate ::SpawnNamed,
+			#( #where_clause_elts, )*
```

Comment:
what's elts?

Reply:
Elements, meaning it is a vector of clauses.